File: Rasha_Toine/Rasha/QLForm/src/ast/expression/LiteralExpression.java
package ast.expression;

import visiting.ExpressionVisitor;
import ast.literal.Literal;

public class LiteralExpression extends Expression {

	private Literal literal;
	
	public LiteralExpression(Literal literal) {
		this.setLiteral(literal);
	}

	@Override
	public <T, U> T accept(ExpressionVisitor<T, U> visitor, U ctx) {
		return visitor.visit(this, ctx);
	}

	public Literal getLiteral() {
		return literal;
	}

	public void setLiteral(Literal literal) {
		this.literal = literal;
	}
}

File: DogeQL/src/main/kotlin/ql/ast/node/expression/LiteralExpression.kt
package ql.ast.node.expression

import ql.ast.location.SourceLocation
import ql.data.value.BaseSymbolValue
import ql.visitor.QuestionnaireASTBaseVisitor

class LiteralExpression(val value: BaseSymbolValue, sourceLocation: SourceLocation) : Expression(sourceLocation) {

    override fun <T> accept(visitor: QuestionnaireASTBaseVisitor<T>): T {
        return visitor.visit(this)
    }

}